types/types.go

```diff
@@ -91,14 +91,29 @@ type ImageReference interface {
 	DeleteImage(ctx context.Context, sys *SystemContext) error
 }
 
+// LayerCompression indicates if layers must be compressed, decompressed or preserved
+type LayerCompression int
+
+const (
+	// PreserveOriginal indicates the layer must be preserved, ie
+	// no compression or decompression.
+	PreserveOriginal LayerCompression = iota
+	// Decompress indicates the layer must be decompressed
+	Decompress
+	// Compress indicates the layer must be compressed
+	Compress
+)
+
 // BlobInfo collects known information about a blob (layer/config).
 // In some situations, some fields may be unknown, in others they may be mandatory; documenting an “unknown” value here does not override that.
```

Comment:
Does the rest of the interface need updating for the presence/validity of `Compression` and `CompressionAlgorithm`?  (I suppose, let’s start with the code and use the documentation as the final check that everything is accounted for.)

Reply:
Yes, let's get the final polish when the code is ready.